noir-projects/noir-protocol-circuits/crates/types/src/hash.nr

```diff
@@ -200,6 +173,10 @@ pub fn pedersen_hash<N>(inputs: [Field; N], hash_index: u32) -> Field {
     dep::std::hash::pedersen_hash_with_separator(inputs, hash_index)
 }
 
+pub fn poseidon_hash<N>(inputs: [Field; N]) -> Field {
+    Poseidon2::hash(inputs, inputs.len())
```

Comment:
Could you try exposing `hash_internal` in the stdlib and calling `Poseidon2::hash_internal(input, N, false)` to see if that gives you a difference?

Reply:
I tried this on a small example and it had no effect.